src/apps/external_executor/external_executor.cpp

```diff
@@ -264,7 +326,7 @@ namespace externalexecutor
                   (http_status)payload.status_code(),
                   std::move(headers),
                   std::move(trailers),
-                  {(const uint8_t*)body_s.data(), body_s.size()}))
+                  {body_s.begin(), body_s.end()}))
```

Comment:
IUIC, this is now constructing a `std::vector` here, rather than a span pointing at the existing payload bytes? That's OK for now because the `std::vector` is moved directly into a buffer in the implementation (so there would already have been a copy there, we've just moved it here). But if we're able to copy directly to the target memory range in this `send_response` call, then we'd prefer to copy _from_ the original byes (with a `span`) than this intermediate `vector`.

I think we should prefer spans in general, and only do copies to an owning `std::vector` internally/as late as possible. That makes it possible to eventually remove intermediate copies entirely, whereas `std::vector`s in the API suggest a requirement for these intermediate copies.

Reply:
Good point. I've changed this back to a `std::span`, which is now also used for streaming data too.